WordPress/gutenberg
# [RNMobile] Implement recovery option for invalid blocks

Related PRs:

* `gutenberg-mobile`: https://github.com/wordpress-mobile/gutenberg-mobile/pull/4996

## What?

If a block's validation fails within the mobile app, users are shown an error but not provided with any options for recovering the block. This PR introduces an option for users to attempt block recovery. They'll be prompted to tap on a block if they wish to attempt recovery.

## Why?

The option to attempt block recovery already exists on the web, as shown in the following screenshot:

<img width="875" alt="Screenshot 2022-06-27 at 23 02 07" src="https://user-images.githubusercontent.com/2998162/176043281-12acd874-72c1-4010-bdd7-33f21dd0494b.png">

This PR, therefore, brings the native behaviour more inline with the web. In doing so, it will provide users with an option to recover content and the look/feel of their content, without them needing to manually re-add blocks/content.

## How?

The `attemptBlockRecovery` function has been implemented following the same patterns found in web's version of the function here: 
https://github.com/WordPress/gutenberg/blob/286e6be2b13fc6ab2059b66d5c79728d811562ee/packages/block-editor/src/components/block-list/block-invalid-warning.js#L122-L124

## Testing Instructions

* Navigate to the post editor in the iOS or Android app.
* Add a new block, making sure to fill out some content or upload media if appropriate.
* Switch the editor to HTML mode.
* Remove the `wp-block-xxx` class name from the block's HTML.
* Switch the editor back to visual mode.
* Verify that the new "Tap to attempt block recovery" prompt appears alongside the now-invalid block.
* Tap on the block and confirm that it's recovered, with the content that was added also being recovered.

## Screenshots or screencast 

| Before  | After |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/176046151-b6076e19-edee-47c7-89fc-f0149f7a2fcb.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/176046423-83c5f57d-2715-4236-a291-8e2eefef4953.png" width="100%"> |

The following recording shows what happens before and after tapping the updated error message: 

https://user-images.githubusercontent.com/2998162/176045857-366250cf-374c-452c-9e66-4548b988202a.mp4
